padertorch/ops/stft_module.py

```diff
@@ -0,0 +1,270 @@
+                pad_width = (pad_width, pad_width)
+            x = F.pad(x, pad_width, mode='constant')
+
+        if self.pad:
+            if x.shape[-1] < length:
+                pad_size = length - x.shape[-1]
+                x = F.pad(x, (0, pad_size))
+            elif stride != 1 and (x.shape[-1] + stride - length) % stride != 0:
+                pad_size = stride - ((x.shape[-1] + stride - length) % stride)
+                x = F.pad(x, (0, pad_size))
+
+        x = torch.unsqueeze(x, 1) # [..., 1, T]
+        weights = self.stft_kernel.to(x)
+        encoded = F.conv1d(x, weight=weights, stride=stride)
+
+        encoded = encoded.view(*org_shape[:-1], *encoded.shape[-2:])
+        encoded = rearrange(encoded, '... feat frames -> ... frames feat')
+        encoded = torch.chunk(encoded, 2, dim=-1)
+        if self.complex_representation == 'stacked':
+            encoded = torch.stack(encoded, dim=-1)
+        elif self.complex_representation == 'concat':
+            encoded = torch.cat(encoded, dim=-1)
+        else:
+            raise NotImplementedError
+
+        return encoded
+
+    def inverse(self, stft_signal):
+        """
+        Args:
+            stft_signal:
+                if complex_representation == 'complex':
+                        shape: [..., frames, num_fbins]
+                if complex_representation == 'stacked':
+                        shape: [..., frames, num_fbins, 2]
+                if complex_representation == 'concat':
+                        shape: [..., frames, 2 * num_fbins]
+            num_samples, list or tensor of #samples
+        Returns:
+            [..., T]
+
+        >>> stft_signal = torch.rand((2, 4, 10, 514))
+        >>> torch_stft = STFT(512, 20, window_length=40, \
+                              complex_representation='concat')
+        >>> torch_signal = torch_stft.inverse(stft_signal)
+        >>> torch_signal.shape
+        torch.Size([2, 4, 180])
+        >>> from paderbox.transform import istft
+        >>> signal_np = stft_signal.numpy()
+        >>> complex_signal = signal_np[..., :257] + 1j* signal_np[..., 257:]
+        >>> time_signal = istft(complex_signal, 512, 20, window_length=40)
+        >>> np.testing.assert_allclose(torch_signal, time_signal, atol=1e-5)
+        """
+
+        org_shape = stft_signal.shape
+        x = stft_signal.view(-1, *org_shape[-2:])
+
+        x = rearrange(x, '... frames feat -> ... feat frames')
+        if self.complex_representation == 'stacked':
+            signal_real, signal_imag = torch.chunk(stft_signal, 2, dim=-1)
+        elif self.complex_representation == 'concat':
+            signal_real, signal_imag = torch.chunk(x, 2, dim=-2)
+        else:
+            raise NotImplementedError
+
+        signal_real = torch.cat(
+            [signal_real, signal_real[:, 1:-1].flip(1)], dim=1)
+        signal_imag = torch.cat(
+            [signal_imag, -signal_imag[:, 1:-1].flip(1)], dim=1)
```

Comment:
Does this assume that stft size is even?

Reply:
yes, you are right either I have to change it to allow uneven sizes as well or add an assert

Reply:
I think an assert here would be enough. We can add support for uneven stft sizes when we find an use case.

Reply:
done